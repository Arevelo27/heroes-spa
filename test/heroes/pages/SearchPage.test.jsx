import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { SearchPage } from "../../../src/heroes/pages/SearchPage";

jest.mock('query-string', () => ({
    parse: jest.fn(() => ({ q: 'batman' })),
}));

describe('Pruebas en <SearchPage />', () => {

    test('debe de mostrarse correctamente con valores por defecto', () => {

        const { container } = render(
            <MemoryRouter initialEntries={['/search']} future={{ v7_startTransition: true, v7_relativeSplatPath: true }}>
                <SearchPage />
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar a Batman y el input con el valor del queryString', () => {

        render(
            <MemoryRouter initialEntries={['/search?q=batman']} future={{ v7_startTransition: true, v7_relativeSplatPath: true }}>
                <SearchPage />
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        expect(input.value).toBe('batman');

        const img = screen.getByRole('img');
        expect(img.src).toContain('/assets/heroes/dc-batman.jpg');

        const alert = screen.getByLabelText('alert-danger');
        expect(alert.style.display).toBe('none');

    });
})  
