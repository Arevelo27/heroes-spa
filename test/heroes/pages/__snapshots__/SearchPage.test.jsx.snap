// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search..."
          type="text"
          value="batman"
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
        style="display: none;"
      >
        Search a hero
      </div>
      <div
        aria-label="alert-danger"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No results found hero 
        <p>
          batman
        </p>
      </div>
      <div
        class="col animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-4"
            >
              <img
                alt="Batman"
                class="card-img"
                src="../assets/heroes/dc-batman.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Batman
                </h5>
                <p
                  class="card-text"
                >
                  Bruce Wayne
                </p>
                <p
                  class="card-text"
                >
                  <small
                    class="text-muted"
                  >
                    Detective Comics #27
                  </small>
                </p>
                <p
                  class="card-text"
                >
                  DC Comics
                </p>
                <a
                  href="/hero/dc-batman"
                >
                  MaÃÅs...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
